tests/test_file.py

```diff
@@ -145,6 +145,16 @@ def test_local_temporary(self):
             self.assertTrue(tmp.exists())
         self.assertFalse(tmp.exists(), f"uri: {tmp}")
 
+        with ResourcePath.temporary_uri(suffix=".yaml", delete=False) as tmp:
+            tmp.write(b"1234")
+            self.assertTrue(tmp.exists(), f"uri: {tmp}")
+        exists = tmp.exists()
+        try:
+            tmp.remove()
+        except Exception:
```

Comment:
It might be better not to swallow all exceptions here... isn't part of the contract of a temporary file that it should be deletable?

Also I found it slightly confusing that the assertion for exists came after the call to `tmp.remove()` -- at first glance this test seemed to be stating that `tmp.remove()` is supposed to fail and the file should still exist at the end.

Reply:
Right. Thanks. I was in a rush and messed up. I realized that if the exists assert fails the clean up line wouldn't run so I did the check outside the assert, but of course, if the exist fails it means that the file isn't there to clean up anyhow so it's not a problem. I'll fix.